diff --git a/stat/stat b/stat/stat
index 8a53b50..be72268 100644
--- a/stat/stat
+++ b/stat/stat
@@ -121,5 +121,18 @@ class Stat: public GS__
 #include "uid.f"
 
 } // FBB
+
+/*
+ *  PATH_MAX is defined by the POSIX standard.  Most modern systems
+ *  provide it.  Older systems define MAXPATHLEN in <sys/param.h> instead.
+ *  hurd-i386 does not provide it...
+ */
+#ifndef PATH_MAX
+#  ifdef MAXPATHLEN
+#    define PATH_MAX	MAXPATHLEN
+#  else /* MAXPATHLEN */
+#    define PATH_MAX	2048
+#  endif /* MAXPATHLEN */
+#endif /* ! PATH_MAX */
         
 #endif
