Index: bobcat-3.18.01/icmake/special
===================================================================
--- bobcat-3.18.01.orig/icmake/special	2013-11-13 05:44:52.000000000 -0500
+++ bobcat-3.18.01/icmake/special	2013-11-26 09:05:11.055508314 -0500
@@ -78,7 +78,7 @@
 
     if (all || gets() == "y")
     {
-        g_sharedLibReq += " -lssl -lcrypto";
+        g_sharedLibReq += " -lssl -lcrypto -lpthread";
         addClasses("SSLCLASSES");
     }    
 }
