--- a/icmake/install	2012-02-08 20:56:04.000000000 +0000
+++ b/icmake/install	2013-09-08 17:32:26.867701666 +0000
@@ -10,7 +10,7 @@
     for (idx = sizeof(man); idx--; )
     {
         file = element(idx, man);
-        run("gzip -9 < " + src + file + " > " + dest + file + ".gz");
+        run("gzip -n -9 < " + src + file + " > " + dest + file + ".gz");
     }
 }
 
@@ -26,7 +26,7 @@
     for (idx = sizeof(files); idx--; )
     {
         file = element(idx, files);
-        run("gzip -9 < " + file + " > " + dest + file + ".gz");
+        run("gzip -n -9 < " + file + " > " + dest + file + ".gz");
     }
 }
     
@@ -59,7 +59,7 @@
 
     rungzip9("tmp/man/man3/", dev + MAN + "/man3/");
     
-    run("gzip -9 < tmp/man/man7/bobcat.7 > " +
+    run("gzip -n -9 < tmp/man/man7/bobcat.7 > " +
                    dev + MAN + "/man7/bobcat.7.gz");
 #endif
 
