--- a/documentation/man/isymcryptstream.yo
+++ b/documentation/man/isymcryptstream.yo
@@ -28,7 +28,7 @@
 using 128 bit sized keys and blocks using `cbc' mode (see below for an
 explanation).
 
-    Refer to the bf(isymcryptstreambuf)(3bobcat) man-page for a descryption of
+    Refer to the bf(isymcryptstreambuf)(3bobcat) man-page for a description of
 available encryption methods.
 
 includefile(include/namespace)
--- a/documentation/man/pattern.yo
+++ b/documentation/man/pattern.yo
@@ -226,7 +226,7 @@
 segmentation fault. This is a well-known problem with static data, see, e.g.,
 tt(http://www.parashift.com/c++-faq-lite/ctors.html#faq-10.15). In situations
 like this prefer the use of a (shared, unique) pointer to a tt(Pattern),
-initialzing the pointer when, e.g., first used.
+initializing the pointer when, e.g., first used.
 
 includefile(include/trailer)
 
